tmva/tmva/src/MethodBase.cxx

```diff
@@ -3039,7 +3039,12 @@ void TMVA::MethodBase::MakeClass( const TString& theClassFileName ) const
    fout << "   virtual ~IClassifierReader() {}" << std::endl;
    fout << std::endl;
    fout << "   // return classifier response" << std::endl;
-   fout << "   virtual double GetMvaValue( const std::vector<double>& inputValues ) const = 0;" << std::endl;
+   if(GetAnalysisType() == Types::kMulticlass) {
+      fout << "   virtual std::vector<Float_t>& GetMulticlassValues( const std::vector<double>& inputValues ) const = 0;" << std::endl;
```

Comment:
This should return the vector by value. Basically the compiler does return value optimisation for this kind of function so there should in practise be no performance hit on this, and we don't have to worry about managing memory.

Reply:
done